examples/cfd/aero_graph_net/inference.py

```diff
@@ -0,0 +1,181 @@
+# limitations under the License.
+
+import logging
+from pathlib import Path
+
+from dgl import DGLGraph
+from dgl.dataloading import GraphDataLoader
+
+import hydra
+from hydra.utils import instantiate, to_absolute_path
+
+import numpy as np
+import pyvista as pv
+import torch
+
+from omegaconf import DictConfig
+
+from modulus.distributed.manager import DistributedManager
+from modulus.launch.utils import load_checkpoint
+
+from loggers import init_python_logging
+from utils import batch_as_dict
+
+
+logger = logging.getLogger("agnet")
+
+
+def dgl_to_pyvista(graph: DGLGraph):
+    """
+    Converts a DGL graph to a PyVista graph.
+
+    Parameters:
+    -----------
+    graph: DGLGraph
+        The input DGL graph.
+
+    Returns:
+    --------
+    pv_graph:
+        The output PyVista graph.
+    """
+
+    pv_graph = pv.PolyData()
+
+    # Assuming "pos" is in the source graph node data.
+    assert "pos" in graph.ndata, f"pos data does not exist, {graph.ndata.keys()=}"
+    pv_graph.points = graph.ndata["pos"].numpy()
+
+    # Create lines from edges.
+    edges = np.column_stack(graph.edges())
+    lines = np.empty((edges.shape[0], 3), dtype=np.int64)
+    lines[:, 0] = 2
+    lines[:, 1:] = edges
+
+    pv_graph.lines = lines.flatten()
+    pv_graph.point_data["p_pred"] = graph.ndata["p_pred"].numpy()
+    pv_graph.point_data["p"] = graph.ndata["p"].numpy()
+    pv_graph.point_data["wallShearStress_pred"] = graph.ndata[
+        "wallShearStress_pred"
+    ].numpy()
+    pv_graph.point_data["wallShearStress"] = graph.ndata["wallShearStress"].numpy()
+
+    return pv_graph
+
+
+class EvalRollout:
+    """MGN inference with a given experiment."""
+
+    def __init__(self, cfg: DictConfig):
+        self.output_dir = Path(to_absolute_path(cfg.output))
+        logger.info(f"Storing results in {self.output_dir}")
+
+        # set device
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        logger.info(f"Using {self.device} device")
+
+        # instantiate dataset
+        logger.info("Loading the test dataset...")
+        self.dataset = instantiate(cfg.data.test)
+        logger.info(f"Using {len(self.dataset)} test samples.")
+
+        # instantiate dataloader
+        logger.info("Creating the dataloader...")
+        self.dataloader = GraphDataLoader(
+            self.dataset,
+            **cfg.test.dataloader,
+        )
+
+        # instantiate the model
+        logger.info("Creating the model...")
+        self.model = instantiate(cfg.model).to(self.device)
+
+        # enable train mode
+        self.model.eval()
+
+        # load checkpoint
+        load_checkpoint(
+            to_absolute_path(cfg.resume_dir),
+            models=self.model,
+            device=self.device,
+        )
+
+        # instantiate losses.
+        logger.info("Creating the losses...")
+        self.loss = instantiate(cfg.loss)
+
+    @torch.no_grad()
```

Comment:
I tend to prefer using `no_grad` instead of `inference_mode` as the latter had a known issue in the past when it was giving an error (`no_grad` worked in that case). As I understand, that bug has been fixed, but my habit stayed :)
Will change it.